src/parser/type_checking.c

```diff
@@ -18958,6 +18958,12 @@ pt_fold_const_expr (PARSER_CONTEXT * parser, PT_NODE * expr, void *arg)
       return expr;
     }
 
+  /* if dblink query, do not constant fold */
+  if (parser->dblink_remote)
+    {
+      return expr;
+    }
```

Comment:
When is dblink_remote set? It seems to affect the expressions unrelated to dblink pushable terms.

Reply:
It is not easy to determine whether it was a dblink-related expr, so I handled it simply.

Reply:
Does it affect the SELECT query as well? If so, the results of the SELECT query may be inconsistent depending on whether or not the DBLINK spec exists.

Reply:
I added the function to check dblink-related expr.